src/components/History/History.tsx

```diff
@@ -153,33 +172,56 @@ const HistoryItem = ({chat}: {chat: ChatHistoryItem}) => {
         return baseMessage as AssistantMessage;
       }
     }))
+    trackEvent('hl_chat_selected', { 
+      chatId: chat.id,
+      messageCount: messages.length
+    });
   }
 
   const onDeleteChat = async (chat: ChatHistoryItem) => {
     openModal('delete-chat', chat)
+    trackEvent('hl_chat_delete_initiated', { 
+      chatId: chat.id
+    });
   }
 
   return (
     <ContextMenu
       key={`menu-${chat.id}`}
       items={[
-        {label: 'Open Chat', onClick: () => onSelectChat(chat)},
+        {
+          label: 'Open Chat', 
+          onClick: () => {
+            onSelectChat(chat);
+            trackEvent('hl_chat_opened_from_context_menu', { 
+              chatId: chat.id
+            });
```

Comment:
Probably better to have one `Chat Opened` event and send `From Context Menu` as a `context` property or something.

Reply:
Done